src/layers/geojson-layer/geojson-utils.js

```diff
@@ -185,3 +186,84 @@ export function getGeojsonFeatureTypes(allFeatures) {
   }, {});
   return featureTypes;
 }
+
+/**
+ * Parse geojson from string
+ * @param {array} geojson feature object values
+ * @returns {boolean} whether the geometry coordinates has length of 4
+ */
+export function coordHasLength4(samples) {
+  let hasLength4 = true;
+  for (let i = 0; i < samples.length; i += 1) {
+    hasLength4 = !samples[i].geometry.coordinates.find(c => c.length < 4);
+    if (samples[i].geometry.coordinates.find(c => c.length < 4)) {
+      break;
+    }
+  }
+  return hasLength4;
+}
+
+/**
+ * Check whether geojson linestring's 4th coordinate is 1) not timestamp 2) unix time stamp 3) real date time
+ * @param {array} data array to be tested if its elements are timestamp
+ * @returns {string} the type of timestamp: unix/datetime/invalid(not timestamp)
+ */
+
+export function containValidTime(data) {
+  const formattedData = [{...data}];
+  const analyzedType = Analyzer.computeColMeta(formattedData);
+  if (analyzedType.find(d => d.category !== 'TIME')) {
+    return false;
+  } else if (['x', 'X'].includes(analyzedType[0].format)) {
+    return 'unix';
+  }
+  return 'dateTime';
```

Comment:
this function should return `analyzedType` which contain the `type` and `format`